src/cascade/executor/cascade_plan.py

```diff
@@ -227,7 +227,24 @@ def from_epiviz_configuration(cls, locations, settings, args):
             setup_task = []
         else:
             setup_task = [(drill[0], "bundle_setup")]
-        tasks = setup_task + [(drill_location, 0) for drill_location in drill]
+
+        substeps = [
+            "prepare_data",
+            "construct_model",
+        ]
+        if settings.policies.fit_strategy == "fit_fixed_then_fit":
+            substeps.append("initial_guess_from_fit_fixed")
+        substeps.extend([
+            "compute_initial_fit",
+            "compute_draws_from_parent_fit",
+            "save_predictions"
+        ])
+
+        tasks = setup_task + [
+            (drill_location, ("estimate_location", substep))
+            for drill_location in drill
+            for substep in substeps
+        ]
```

Comment:
The line after this, the `task_pairs`, is where this falls down. We need to preserve, in the graph, the dependency tree, so that we can know what can run in parallel.

Reply:
I see now that these are, in fact, linear, so the zip is fine for this specific application. Interesting. The ` `compute_draws` should be the task array, but... does it show up as one job here? I'm currently still confused about that.

Reply:
Yes, they are linear. I haven't touched the problem of how to represent dependency. Nor have I broken compute_draws into individual jobs. It's currently a single job like it's always been. The reason for that is that breaking it up such that we preserve parallelism within a single-node job requires a lot of extra work and I decided that was out of scope.